documentation/rfc/distributed.md

```diff
@@ -12,40 +12,57 @@ This scenario allows us to call multiple engines for computing fallbacks. These
 durations by using the configured streetnetwork engine and its matrix API.
 
-At this point, we can start the public transport part of the algorithm. We will trigger the computation of each valid combination
+6. At this point, we can start the public transport part of the algorithm. We will trigger the computation of each valid combination
 of departure mode and arrival mode. To start it, we need:
   - the origin matrix
   - the destination matrix
   - the direct path
 
-Kraken returns journeys that need to be completed by computing: 
- - a streetnetwork journey from the origin to the departure's stop_point of the first section 
- - another journey from the destination's stop_point of the last section to the destination.
- 
-Like others operations, this is done asynchronously as soon as we get a response from Kraken. If multiple journeys
-use the same fallback journey, it will only be computed one time.
+7. We wait on Kraken to perform all the computation, and recall step 6. if we need more journeys.
+
+8. Now we have the public transport journeys and the stop_point's matrix fallback timings (6. and 5.), so we
+can build temporary crow-flies fallbacks foreach journeys. The crow-flies will have a realistic fallback
+durations but without the actual details (to be computed later on by the street-network service).
+With an accurate approximation of the end-to-end journeys' durations, we can apply all the different filters
+(ie. long waiting, min transfert, direct path etc...).
+ ```
+ Origin     <tmp_crow_fly> 1st stop_point <public_transport> last stop_point <tmp_crow_fly> Destination
+ 19 rue Hec <----5min---->  Gare de Lyon  <=====M1 4min====>  Hotel de Ville <----2min----> Notre-Dame
+```
+
+9. Finaly, we can complete the fallback details from the list of filtered journeys by replacing the temporary crow-fly with the detailed street network. We need to compute:
+ - a streetnetwork journey from the origin to the departure's stop_point of the first section
+ - another streetnetwork journey from the destination's stop_point of the last section to the destination
+```
+ Origin     <street_network> 1st stop_point <public_transport> last stop_point <street_network> Destination
+ 19 rue Hec <--walk 5min---> Gare de Lyon   <=====M1 4min====> Hotel de Ville  <--bike 2min---> Notre-Dame
+```
+
+Like other operations, this is done asynchronously as soon as we get a response from Kraken. If multiple journeys
+use the same fallback journey, it will be computed only once.
 
 For instance, a request like—https://api.navitia.io/v1/coverage/fr-idf/journeys?from=stop_area%3AOIF%3ASA%3A59238&to=2.33071%3B48.83890&first_section_mode%5B%5D=bike&first_section_mode%5B%5D=walking&datetime=20180614T113500&last_section_mode%5B%5D=walking&, will produce the following calls.
 
 Dependency graph:
 
-![graph](../diagrams/distributed.png)
+![graph](../diagrams/distributed_diagram.svg)
```

Comment:
svg output seems worse than png :disappointed: 

When a choice is done and working, it would be nice to add the command (comment in gv file) to regenerate output.

Reply:
How can it be worse ? Svg is vectorial and scale properly to any resolution (when png is rasterised...).

Reply:
Not saying it's a worse format, I agree the format is way better.
Just compare the output between png I sent on slack (proper place for titles) and this svg (too little boxes)...
I'd rather have a clean output but not-so-scalable format than the opposite :)